fhireval/test_suite/search/test_coresearch_rest.py

```diff
@@ -2,20 +2,242 @@
 import pdb
 
 from fhir_walk.model.patient import Patient
+from fhireval.data import prep_server, id_list
 
-test_id = f"{'2.2.1.1':<10} - Core FHIR Search"
+from pprint import pformat
+
+test_id = f"{'2.10.1.1':<10} - Core FHIR Search"
 
 test_weight = 15
 
 # TODO Get IDs for data to be qureied
 # Profiles to be queried
 #   'ResearchStudy', 'ResearchSubject', 'Organization', 'Specimen', 'Observation', 'DiagnosticReport','Condition', 'DocumentReference', 'Task', 'Group'
 
-# This is just data I've been using as fake data. The official dataset will likely
-# look very different
-def test_basic_patient_queries(host):
-    # This tests Patient?identifier='fd-sub1'
-    patient = Patient.PatientBySubjectID(None, 'fd-sub1', host) 
-    assert patient is not None
-    assert patient.sex == 'male'
+# When comparing search results, ignore certain columns that will definitely be 
+# different in the value returned from fhir
+_params_to_ignore = ['meta', 'id']
+# Simple tests for identity
+@pytest.mark.parametrize("resource", ["Patient", "Observation", "Condition", "ResearchSubject", "ResearchStudy"])
+def test_search_by_identifier(host, prep_server, resource):
+    example_resource = prep_server['_ALL'][resource][0]
+    example_system = example_resource['identifier'][0]['system']
+    example_identity = example_resource['identifier'][0]['value']
+
+    # Simple query based on the value
+    # This only works if there isn't a "|" in the identifier, otherwise, that first part will 
+    # be compared to the system. 
+    if "|" not in example_identity:
+        qry_response = host.get(f"{resource}?identifier={example_identity}").entries
+        assert len(qry_response) == 1, f"Only one record returned (got {len(qry_response)}"
+        fhir_record = qry_response[0]
+        
+        for key in example_resource.keys():
+            if key not in _params_to_ignore:
+                if 'resource' not in fhir_record:
+                    pdb.set_trace()
+                assert example_resource[key] == fhir_record['resource'][key], f"Testing for attribute {key}"
+
+    # Simple query based on system|value
+    qry_response = host.get(f"{resource}?identifier={example_system}|{example_identity}").entries
```

Comment:
I've also figured our adding `|` to the unique string identifiers was a bad approach (I replaced it with `-` recently). So, I am not sure which will be better, fixing those IDs (removing `|`) or keeping this conditional. If we do the former, then the query string could simply be like:

```python
query_string = f"{resource}?identifier={example_identity}") if not example_system else f"{resource}?identifier={example_system}|{example_identity}"
```

Reply:
I actually did change all of those identifiers to remove the "|" character, but just in case some data is brought in from other groups that does have it, I've changed the selection to ensure that what we are testing has both a system and no pipes in the first identifier. That allows me to perform both tests (with and without a system) reliably.